packages/taiko-client/driver/driver.go

```diff
@@ -147,10 +148,11 @@ func (d *Driver) InitFromConfig(ctx context.Context, cfg *Config) (err error) {
 
 			log.Info("P2PNode", "Addrs", d.p2pNode.Host().Addrs(), "PeerID", d.p2pNode.Host().ID())
 
-			if !reflect2.IsNil(d.Config.P2PSignerConfigs) {
-				if d.p2pSigner, err = d.P2PSignerConfigs.SetupSigner(d.ctx); err != nil {
-					return err
-				}
+			if reflect2.IsNil(d.Config.P2PSignerConfigs) {
+				return errors.New("missing P2P signer setup")
```

Comment:
If this is necessary for P2P gossip, let's move this check to `config.go`?

Reply:
The `P2PSignerConfigs` is only used when starting the p2p server, maybe checking here can be simpler?

Reply:
I mean in `config.go`, if p2p is enabled, we check this, since its also required, right? Its more like a flag-related issue?

Reply:
Fixed at this [commit](https://github.com/taikoxyz/taiko-mono/pull/18980/commits/cffca250a3b5412c0186c3b2891678461f07f20c).